### PR TITLE
new getTimezones function

### PR DESCRIPTION
**Pull request by jebeck**
Instead of the hard-coded list (with a data structure that didn't play well with the react-select component we're using in the uploader), we generate an object from the list of loaded moment timezones.

The object has a bunch of categories for easily grabbing subsets of timezones of interest - (1) just the four major U.S. timezones, (2) the remainder of the U.S. timezones, (3) all Canadian, European, Australian, and Kiwi timezones, and then (4) the rest of the world.

### CODE REVIEW

File: lib/datetimeWrapper.js [hunk 1]
```diff
@@ -317,9 +307,82 @@ var datetimeWrapper = function(moment) {
...
+        var timezone = {
+          label: format('%s (UTC%s)', tzNames[i], offsetAsString),
+          value: tzNames[i],
+          offset: Number(offsetAsString)
+        };
+
+        if (big4[tzNames[i]] === true) {
+          timezones.bigFour.push(timezone);
+        }
+        else if (tzNames[i].search('US/') !== -1) {
```
Comment (darinkrauss) [inline lib/datetimeWrapper.js]: Might want to make it `== 0` to ensure it is the first character of the string.  (Same for the ones below.)

File: lib/datetimeWrapper.js [hunk 2]
```diff
@@ -317,9 +307,82 @@ var datetimeWrapper = function(moment) {
...
+          timezones.unitedStates.push(timezone);
+        }
+        else if (tzNames[i].search('Canada') !== -1) {
```
Comment (darinkrauss) [inline lib/datetimeWrapper.js]: Might want a trailing slash (like `US/`). (Same for the ones below.)

File: lib/datetimeWrapper.js [hunk 3]
```diff
@@ -317,9 +307,82 @@ var datetimeWrapper = function(moment) {
...
+        var offsetAsString = offsetInHours > 0 ?
+          '-' + offsetInHours : '+' + Math.abs(offsetInHours);
+
+        var timezone = {
+          label: format('%s (UTC%s)', tzNames[i], offsetAsString),
+          value: tzNames[i],
+          offset: Number(offsetAsString)
+        };
+
+        if (big4[tzNames[i]] === true) {
```
Comment (darinkrauss) [inline lib/datetimeWrapper.js]: Can we drop the `=== true` since it is a boolean or is the standard you use?

File: lib/datetimeWrapper.js [hunk 4]
```diff
@@ -317,9 +307,82 @@ var datetimeWrapper = function(moment) {
...
+        // offset for each timezone (i.e., adjust for DST or not)
+        // the offset is part of the `label` attribute
+        var offsetInHours = moment.duration(zone.offset(now), 'minutes')
+          .asMilliseconds()/(1000*60*60);
+
+        // because moment signs offsets the opposite way from the ISO 8601 standard
+        // the standard is negative for N. American timezones
+        // but moment (and JavaScript in general) represent them as positive
+        var offsetAsString = offsetInHours > 0 ?
+          '-' + offsetInHours : '+' + Math.abs(offsetInHours);
```
Comment (darinkrauss) [inline lib/datetimeWrapper.js]: The old code presented offset like ` UTC+09:30`.  Now it presents it as `UTC+9.5`.  Is this what we want?

File: lib/datetimeWrapper.js [hunk 5]
```diff
@@ -317,9 +307,82 @@ var datetimeWrapper = function(moment) {
...
+          timezones.hoisted.push(timezone);
+        }
+        // reformat these a tiny bit so that 'New Zealand' is searchable
+        else if (tzNames[i].search('NZ') !== -1) {
+          timezone.label = timezone.label.replace('NZ', 'New Zealand');
+          timezones.hoisted.push(timezone);
+        }
+        // there's a whole bunch of timezones listed as "Etc/GMT+1"
+        // that clutter things up a lot...
+        else if (tzNames[i].search('GMT') !== -1) {}
```
Comment (darinkrauss) [inline lib/datetimeWrapper.js]: it looks like there are still a bunch of weird ones in there like: `PST8PDT`, `EST`.  Not sure if we want to filter these or not.

**Comment by darinkrauss** [comment]
@jebeck @jh-bate A few comments.  The code ones are minor and are style related.  The two of concern are the format of the UTC string (eg. `UTC+9.5`) and the inclusion of a bunch of weird timezone names.

File: lib/datetimeWrapper.js (hunk 5 above)
Comment (jebeck) [inline lib/datetimeWrapper.js]: We really hope the experience is going to be more people using the dropdown (for the U.S. timezones at the top) or searching for things like 'Canada' or 'New Zealand', so I think for now I'm fine with leaving the other weird ones in.

File: lib/datetimeWrapper.js (hunk 4 above)
Comment (jebeck) [inline lib/datetimeWrapper.js]: It's not easy to format the +09:30 thing with moment - they don't provide formatters for durations. Open to writing the code to do it later, but I asked Brandon if this compromise would be fine for now, and he agreed.

File: lib/datetimeWrapper.js (hunk 3 above)
Comment (jebeck) [inline lib/datetimeWrapper.js]: I've been bitten so badly by truthiness in the past (although mostly with `!`, I suppose) that I'm trying to get out of the habit, but it is rather superfluous here. I may change.

File: lib/datetimeWrapper.js (hunk 2 above)
Comment (jebeck) [inline lib/datetimeWrapper.js]: Ah, good catch! Will do.

File: lib/datetimeWrapper.js (hunk 1 above)
Comment (jebeck) [inline lib/datetimeWrapper.js]: Good point, I'll change it.

**Pull request merged**